feature/bgp/policybase/otg_tests/nested_policies/nested_policies_test.go

```diff
@@ -234,146 +235,220 @@ func TestBGPNestedPolicies(t *testing.T) {
-	// Configure a prefix-set for route filtering/matching.
-	prefixSet := rp.GetOrCreateDefinedSets().GetOrCreatePrefixSet(v4PrefixSet)
-	prefixSet.SetMode(oc.PrefixSet_Mode_IPV4)
-	prefixSet.GetOrCreatePrefix(advertisedIPv41.cidr(t), maskLenExact)
+		t.Logf("Configuring nested policy")
+		// Configure a prefix-set for route filtering/matching.
+		prefixSet := rp.GetOrCreateDefinedSets().GetOrCreatePrefixSet(v4PrefixSet)
+		prefixSet.SetMode(oc.PrefixSet_Mode_IPV4)
+		prefixSet.GetOrCreatePrefix(advertisedIPv41.cidr(t), maskLenExact)
 
-	if !deviations.SkipSetRpMatchSetOptions(dut) {
-		stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+		if !deviations.SkipSetRpMatchSetOptions(dut) {
+			stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetMatchSetOptions(oc.RoutingPolicy_MatchSetOptionsRestrictedType_ANY)
+		}
+		stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetPrefixSet(v4PrefixSet)
+	}
+	if operation == "set" {
+		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
+	} else if operation == "delete" {
+		gnmi.BatchDelete(batch, gnmi.OC().RoutingPolicy().Config())
+		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
 	}
-	stmt2.GetOrCreateConditions().GetOrCreateMatchPrefixSet().SetPrefixSet(v4PrefixSet)
-	gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
-
-	// Configure the nested policy.
 	dni := deviations.DefaultNetworkInstance(dut)
-	rpPolicy := root.GetOrCreateRoutingPolicy()
-	statPath := rpPolicy.GetOrCreatePolicyDefinition(v4LPPolicy).GetStatement(v4LPStatement).GetOrCreateConditions()
-	statPath.SetCallPolicy(v4PrefixPolicy)
-	gnmi.Update(t, dut, gnmi.OC().RoutingPolicy().Config(), rpPolicy)
-
+	if operation == "set" {
+		// Configure the nested policy.
+		rpPolicy := root.GetOrCreateRoutingPolicy()
+		statPath := rpPolicy.GetOrCreatePolicyDefinition(v4LPPolicy).GetStatement(v4LPStatement).GetOrCreateConditions()
+		statPath.SetCallPolicy(v4PrefixPolicy)
+		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rpPolicy)
+	} else if operation == "delete" {
+		gnmi.BatchDelete(batch, gnmi.OC().RoutingPolicy().Config())
+		gnmi.BatchReplace(batch, gnmi.OC().RoutingPolicy().Config(), rp)
+	}
 	// Configure the parent BGP import policy.
-	path := gnmi.OC().NetworkInstance(dni).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).Bgp().Neighbor(atePort1.IPv4).AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).ApplyPolicy()
-	policy := root.GetOrCreateNetworkInstance(dni).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).GetOrCreateBgp().GetOrCreateNeighbor(atePort1.IPv4).GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).GetOrCreateApplyPolicy()
-	policy.SetImportPolicy([]string{v4LPPolicy})
-	if !deviations.RoutingPolicyChainingUnsupported(dut) {
-		gnmi.Update(t, dut, path.Config(), policy)
+	if deviations.RoutePolicyUnderAFIUnsupported(dut) {
```

Comment:
Multiple time if operation == "set" / if operation == "delete"  is used
	Please clean it up through the entire changes
	
	if deviations.RoutePolicyUnderAFIUnsupported(dut) {
		//policy under peer group
		path := gnmi.OC().NetworkInstance(dni).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).Bgp().PeerGroup(peerGrpNamev4).ApplyPolicy()
		policy := root.GetOrCreateNetworkInstance(dni).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).GetOrCreateBgp().GetOrCreatePeerGroup(peerGrpNamev4).GetOrCreateApplyPolicy()
		policy.SetImportPolicy([]string{v4LPPolicy})
		if operation == "set" {
			gnmi.BatchReplace(batch, path.Config(), policy)
		} else if operation == "delete" {
			gnmi.BatchDelete(batch, path.Config())
		}
	} else {
		path := gnmi.OC().NetworkInstance(dni).Protocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).Bgp().Neighbor(atePort1.IPv4).AfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).ApplyPolicy()
		policy := root.GetOrCreateNetworkInstance(dni).GetOrCreateProtocol(oc.PolicyTypes_INSTALL_PROTOCOL_TYPE_BGP, bgpName).GetOrCreateBgp().GetOrCreateNeighbor(atePort1.IPv4).GetOrCreateAfiSafi(oc.BgpTypes_AFI_SAFI_TYPE_IPV4_UNICAST).GetOrCreateApplyPolicy()
		policy.SetImportPolicy([]string{v4LPPolicy})
		if operation == "set" {
			if !deviations.RoutingPolicyChainingUnsupported(dut) {
				gnmi.BatchUpdate(batch, path.Config(), policy)
			} else {
				gnmi.BatchReplace(batch, path.Config(), policy)
			}
		} else if operation == "delete" {
			gnmi.BatchDelete(batch, path.Config())
		}
	}

Reply:
Updated the code to reduce multiple if else condition.